qa-tests-backend/src/main/groovy/orchestratormanager/Kubernetes.groovy

```diff
@@ -1393,23 +1399,23 @@ class Kubernetes implements OrchestratorMain {
 
     List<K8sRole> getRoles() {
         return evaluateWithRetry(1, 2) {
-            def roles = []
+            List<K8sRole> roles = []
             client.rbac().roles().inAnyNamespace().list().items.each {
                 roles.add(new K8sRole(
                         name: it.metadata.name,
                         namespace: it.metadata.namespace,
                         clusterRole: false,
                         labels: it.metadata.labels ? it.metadata.labels : [:],
                         annotations: it.metadata.annotations ? it.metadata.annotations : [:],
-                        rules: it.rules.collect {
+                        rules: it.rules ? it.rules.collect {
```

Comment:
```suggestion
                        rules: it?.rules.collect {
```

https://ideone.com/ZBHa6e

Reply:
FTR the `K8sRbacTest` error on [this line](https://github.com/stackrox/stackrox/commit/df2b02e9c23343c5d29f8eb56b7ecd4ccd327ca3#diff-2a047800df52fb14fa11cf03ef5862df577faf528e5d9e857329dd913945b84cR1406) [was](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/stackrox_stackrox/5082/pull-ci-stackrox-stackrox-master-gke-qa-e2e-tests/1632730807896379392):
```
	at K8sRbacTest.Verify scraped roles(K8sRbacTest.groovy:168)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Iterable.iterator()" because "self" is null
	at orchestratormanager.Kubernetes.getRoles_closure49$_closure101(Kubernetes.groovy:1406)
	at orchestratormanager.Kubernetes.getRoles_closure49(Kubernetes.groovy:1399)
	at orchestratormanager.Kubernetes.getRoles_closure49(Kubernetes.groovy)
	at util.Helpers.evaluateWithRetry(Helpers.groovy:27)
	at orchestratormanager.Kubernetes.getRoles(Kubernetes.groovy:1397)
	at K8sRbacTest.Verify scraped roles_closure9(K8sRbacTest.groovy:170)
	at K8sRbacTest.Verify scraped roles_closure9(K8sRbacTest.groovy)
	at util.Helpers.evaluateWithRetry(Helpers.groovy:27)
	at util.Helpers.withRetry(Helpers.groovy:31)
	... 1 more
```

I took this to mean that (in the `@CompileStatic` world?) it's not OK for `rules` to be `null`. I introduced the ternary operator and the error disappeared. Your suggestion seems to be guarding against `it` being `null`. But if that was the case, then the failure would be even earlier, when `it.metadata` is being dereferenced. So I don't think this suggestion is all that useful :thinking:

Reply:
Seems the problem now is in nearly identical code around line 1530.. let me try fixing that...